examples/mnist/readme.md

```diff
@@ -27,55 +25,74 @@ static const bool tbl [] = {
 #undef X
 
 // construct nets
-nn << convolutional_layer<tan_h>(32, 32, 5, 1, 6)  // C1, 1@32x32-in, 6@28x28-out
-    << average_pooling_layer<tan_h>(28, 28, 6, 2)   // S2, 6@28x28-in, 6@14x14-out
-    << convolutional_layer<tan_h>(14, 14, 5, 6, 16,
-            connection_table(tbl, 6, 16))            // C3, 6@14x14-in, 16@10x10-in
-    << average_pooling_layer<tan_h>(10, 10, 16, 2)  // S4, 16@10x10-in, 16@5x5-out
-    << convolutional_layer<tan_h>(5, 5, 5, 16, 120) // C5, 16@5x5-in, 120@1x1-out
-    << fully_connected_layer<tan_h>(120, 10);       // F6, 120-in, 10-out
+//
+// C : convolution
+// S : sub-sampling
+// F : fully connected
+nn << convolutional_layer<tan_h>(32, 32, 5, 1, 6,  // C1, 1@32x32-in, 6@28x28-out
+        padding::valid, true, 1, 1, backend_type)
+   << average_pooling_layer<tan_h>(28, 28, 6, 2)   // S2, 6@28x28-in, 6@14x14-out
+   << convolutional_layer<tan_h>(14, 14, 5, 6, 16, // C3, 6@14x14-in, 16@10x10-in
+        connection_table(tbl, 6, 16),
+        padding::valid, true, 1, 1, backend_type)
+   << average_pooling_layer<tan_h>(10, 10, 16, 2)  // S4, 16@10x10-in, 16@5x5-out
+   << convolutional_layer<tan_h>(5, 5, 5, 16, 120, // C5, 16@5x5-in, 120@1x1-out
+        padding::valid, true, 1, 1, backend_type)
+   << fully_connected_layer<tan_h>(120, 10,        // F6, 120-in, 10-out
+        true, backend_type)
 ```
 
-What does ```tbl``` mean? LeNet has "sparsity" between S2 and C3 layer. Specifically, each feature map in C3 is connected to a subset of S2's feature maps so that each feature maps get different set of inputs (and hopefully they become compelemtary feature extractor).
-tiny-dnn supports this sparsity by ```connection_table``` structure which parameters of constructor are ```bool``` table and number of in/out feature maps.
+What does ```tbl``` mean? LeNet has "sparsity" between S2 and C3 layer. Specifically, each feature map in C3 is connected to a subset of S2's feature maps so that each of the feature maps get different set of inputs (and hopefully they become compelemtary feature extractor).
+Tiny-dnn supports this sparsity by ```connection_table``` structure which parameters of constructor are ```bool``` table and number of in/out feature maps.
 
 ## Loading Dataset
-Tiny-cnn supports idx format, so all you have to do is calling parse_mnist_images and parse_mnist_labels functions.
+Tiny-dnn supports MNIST idx format, so all you have to do is calling parse_mnist_images and parse_mnist_labels functions.
 
 ```cpp
 // load MNIST dataset
 std::vector<label_t> train_labels, test_labels;
 std::vector<vec_t> train_images, test_images;
 
-parse_mnist_labels("train-labels.idx1-ubyte", &train_labels);
-parse_mnist_images("train-images.idx3-ubyte", &train_images, -1.0, 1.0, 2, 2);
-parse_mnist_labels("t10k-labels.idx1-ubyte", &test_labels);
-parse_mnist_images("t10k-images.idx3-ubyte", &test_images, -1.0, 1.0, 2, 2);
+parse_mnist_labels(data_dir_path+"/train-labels.idx1-ubyte",
+                   &train_labels);
```

Comment:
yes use `static_cast<cnn_size_t>`

@nyanp something to say about 64bit target?